src/plugins/mer/merbuildconfigurationaspect.cpp

```diff
@@ -120,6 +129,63 @@ class MerBuildConfigurationWidget : public ProjectExplorer::NamedWidget
         vbox->addWidget(m_detailsContainer);
 
         updateSummary();
+
+        connect(m_ui->signPackagesCheckBox, &QCheckBox::toggled, [&](bool checked) {
+            m_aspect->setSignPackagesEnabled(checked);
+            m_ui->signPackageKeyLineEdit->setEnabled(checked);
+            m_ui->signPackageKeyPassFileChooser->setEnabled(checked);
+        });
+        connect(m_ui->signPackageKeyLineEdit, &QLineEdit::textChanged, [&](const QString &keyName) {
+            m_aspect->setSignPackageKey(keyName);
+        });
+        connect(m_ui->signPackageKeyPassFileChooser, &Utils::PathChooser::pathChanged, [&](const QString &passFilePath) {
+            m_aspect->setSignPackagePassFile(passFilePath);
+        });
+
+        Kit *kit = m_aspect->buildConfiguration()->buildSystem()->kit();
+        BuildEngine *engine = MerSdkKitAspect::buildEngine(kit);
+
+        const QString signKeysFolder = QDir::toNativeSeparators(engine->sharedConfigPath().toString())
+                                       + Sfdk::Constants::BUILD_ENGINE_HOST_GNUPG_PATH_POSTFIX;
+        m_ui->signPackageKeyLineEdit->setText(m_ui->signPackageKeyLineEdit->text().arg(signKeysFolder));
+
+        QStringList keys;
+        QDir keysDir(signKeysFolder);
+        for (const auto &keyInfo : keysDir.entryInfoList({PACKAGE_SIGN_KEY_FILE_FILTER}, QDir::Files))
+            keys << keyInfo.completeBaseName();
+
+        auto completer = new QCompleter(keys, this);
+        completer->setCaseSensitivity(Qt::CaseInsensitive);
+        m_ui->signPackageKeyLineEdit->setCompleter(completer);
+
+        m_ui->signPackageKeyPassFileChooser->setExpectedKind(PathChooser::Kind::File);
+        m_ui->signPackageKeyPassFileChooser->setPromptDialogTitle(tr("Select the passphrase file for the sign key"));
+        const QString toolTip = tr("Passphrase file for the sign key");
+        m_ui->signPackageKeyPassFileChooser->lineEdit()->setPlaceholderText(toolTip);
+        m_ui->signPackageKeyPassFileChooser->setToolTip(toolTip);
```

Comment:
`signKeysFolder` is a location internal to the SDK. It is not a place the user should use as a primary storage for their keys - it will be removed during SDK uninstallation. The user should keep their keys stored on the regular GPG keyring and then:

> When the package.signing-user <name> option is set, sfdk should ensure that the corresponding private key is exported as ~/SailfishSDK/vmshare/gnupg/<name>.key prior to executing a command on the build engine.

(quoted from sailfishos/sdk-setup#179)

So what we should offer here to choose from is what we get with `gpg --list-secret-keys`. I would start with a combo box with items in formatted as `NAME <EMAIL> [SHORT_ID]`.

I think we have no other option than to pass the ID as `package.signing-user` (which should accept exactly the same values as gpg's `--local-user` option). NAME+EMAIL may not be unique. But it would be more user friendly when seen in build output, so it would be great if all of ID+NAME+EMAIL could be combined in argument to gpg's `--local-user` option somehow. The documentation does not mention the details, the implementation does not seem to allow that after brief check - maybe you spot a possibility there https://github.com/gpg/gnupg/blob/357ad9ae29677c1676b56d2b81282e2f78ec8040/common/userids.c#L41

Reply:
So in sfdk we have to call `gpg --export-secret-keys <key-name>` on the developer host and then use it?

Reply:
yup

Reply:
Hm, on Mac OS no `gpg`  by default and on Windows it's will be avaliabel only if running QtCreator under GitBash

Reply:
I wouldn't see it as granted that gpg is available even on Linux by default. This is not an issue. Want to gpg sign? - Install gpg first! :)

On Windows you mean it is not in PATH I believe. We can locate it the same way as we do for SSH - sitting in the same dir as git: https://github.com/sailfishos/sailfish-qtcreator/blob/master/src/plugins/projectexplorer/projectexplorer.cpp#L1992

Reply:
Add search git location because gpg installed with git bash